components/list/list-item-generic.js

```diff
@@ -0,0 +1,80 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+export class ListItemGeneric extends RtlMixin(LitElement) {
+
+	static get styles() {
+		return css`
+			:host {
+				display: grid;
+				grid-template-columns:
+					[start outside-control-start] minmax(0, min-content)
+					[control-start outside-control-end] minmax(0, min-content)
+					[control-end content-start] auto
+					[content-end actions-start] auto
+					[end actions-end];
+			}
+			::slotted([slot="outside-control"]),
+			::slotted([slot="control"]),
+			::slotted([slot="content"]),
+			::slotted([slot="actions"]) {
+				grid-row: 1 / 2;
+			}
+			::slotted([slot="outside-control"]) {
+				grid-column: outside-control-start / outside-control-end;
+			}
+
+			::slotted([slot="control"]) {
```

Comment:
I was thinking that this would be `::slotted(.d2l-list-item-area-control)` or something similar. So anything with that class gets place in the slot. Let me know if this would cause other problems that I do not see.

Reply:
I could see this causing the problems for focusing? Which might make slots better?

Reply:
So yeah, this was sort of my rational for doing it this way. This means that no matter what order the user defines their areas in, the focus order is always the same because the DOM structure is the same.

Reply:
Would allowing the user define the focus order be good? I'm not sure. Since it as easy as place in order of how you want focus to work it might be good. The user in this case will be new list items renderers which should be very few and far between.

Reply:
If the user is able to define their own focus order, doesn't that sort of make [US115380](https://rally1.rallydev.com/#/detail/userstory/382436537132?fdp=true) a moot point?

The way we've written these, it means users can only define focus order within action areas, but the action areas themselves have a predefined focus order.

Reply:
Yup! you are right :)

Reply:
So I guess it's about how much control do we want to give here: more v less. I'm fine with either approach TBH but I think giving less forces an accessibility standard that's good, so I would argue forcing focus order is better in that case.